client/core/trade.go

```diff
@@ -133,209 +132,72 @@ type trackedTrade struct {
 	order.Order
 	// mtx protects all read-write fields of the trackedTrade and the
 	// matchTrackers in the matches map.
-	mtx                sync.RWMutex
-	metaData           *db.OrderMetaData
-	dc                 *dexConnection
-	db                 db.DB
-	latencyQ           *wait.TickerQueue
-	wallets            *walletSet
-	preImg             order.Preimage
-	csum               dex.Bytes // the commitment checksum provided in the preimage request
-	mktID              string
-	coins              map[string]asset.Coin
-	coinsLocked        bool
-	lockTimeTaker      time.Duration
-	lockTimeMaker      time.Duration
-	change             asset.Coin
-	changeLocked       bool
-	cancel             *trackedCancel
-	matches            map[order.MatchID]*matchTracker
-	notify             func(Notification)
-	formatDetails      func(Topic, ...interface{}) (string, string)
-	epochLen           uint64
-	fromAssetID        uint32
-	options            map[string]string
-	redemptionReserves uint64
-	redemptionLocked   uint64
-	refundReserves     uint64
-	refundLocked       uint64
-}
-
-// newTrackedTrade is a constructor for a trackedTrade.
-func newTrackedTrade(dbOrder *db.MetaOrder, preImg order.Preimage, dc *dexConnection, epochLen uint64,
+	mtx           sync.RWMutex
+	metaData      *db.OrderMetaData // includes lotSize, maxFeeRate, etc.
+	dc            *dexConnection
+	db            db.DB
+	latencyQ      *wait.TickerQueue
+	wallets       *walletSet
+	preImg        order.Preimage
+	csum          dex.Bytes // the commitment checksum provided in the preimage request
+	mktID         string
+	coins         map[string]asset.Coin // funding coins, not redeem/refund reserves
+	coinsLocked   bool
+	lockTimeTaker time.Duration
+	lockTimeMaker time.Duration
+	change        asset.Coin
+	changeLocked  bool
+	cancel        *trackedCancel
+	matches       map[order.MatchID]*matchTracker
+	notify        func(Notification)
+	formatDetails func(Topic, ...interface{}) (string, string)
+	epochLen      uint64
+	fromAssetID   uint32
+	options       map[string]string
+
+	// For assets requiring extra funds to create the redeem and refund
+	// transactions, the reserved quantities are tracked.
+	redeemRes *reservesCounter
+	refundRes *reservesCounter
+}
+
+// newTrackedTrade is a constructor for a trackedTrade. The dexConnection must
+// have a valid config.
+func newTrackedTrade(dbOrder *db.MetaOrder, preImg order.Preimage, dc *dexConnection,
```

Comment:
Possible simplification to extract from this PR. (`epochLen` is in the `dc`, so a redundant arg)